PROTO_tests/tests/dataframe_test.py

```diff
@@ -0,0 +1,704 @@
+        assert_frame_equal(ref_df, glued.to_pandas())
+
+        df_keyerror = self.build_ak_keyerror()
+        with pytest.raises(KeyError):
+            ak.DataFrame.concat([df, df_keyerror])
+
+        df_typeerror = self.build_ak_typeerror()
+        with pytest.raises(TypeError):
+            ak.DataFrame.concat([df, df_typeerror])
+
+    def test_head(self):
+        df = self.build_ak_df()
+        ref_df = self.build_pd_df()
+
+        hdf = df.head(3)
+        hdf_ref = ref_df.head(3).reset_index(drop=True)
+        assert_frame_equal(hdf_ref, hdf.to_pandas())
+
+    def test_tail(self):
+        df = self.build_ak_df()
+        ref_df = self.build_pd_df()
+
+        tdf = df.tail(2)
+        tdf_ref = ref_df.tail(2).reset_index(drop=True)
+        assert_frame_equal(tdf_ref, tdf.to_pandas())
+
+    def test_groupby_standard(self):
+        df = self.build_ak_df()
+        gb = df.GroupBy("userName")
+        keys, count = gb.count()
+        assert keys.to_list() == ["Bob", "Alice", "Carol"]
+        assert count.to_list() == [2, 3, 1]
+        assert gb.permutation.to_list() == [1, 4, 0, 2, 5, 3]
+
+        gb = df.GroupBy(["userName", "userID"])
+        keys, count = gb.count()
+        assert len(keys) == 2
+        assert keys[0].to_list() == ["Carol", "Bob", "Alice"]
+        assert keys[1].to_list() == [333, 222, 111]
+        assert count.to_list() == [1, 2, 3]
+
+        # testing counts with IPv4 column
+        s = ak.DataFrame({"a": ak.IPv4(ak.arange(1, 5))}).groupby("a").count()
+        pds = pd.Series(
+            data=np.ones(4, dtype=np.int64),
+            index=pd.Index(data=np.array(["0.0.0.1", "0.0.0.2", "0.0.0.3", "0.0.0.4"], dtype="<U7")),
+        )
+        assert_series_equal(pds, s.to_pandas())
+
+        # testing counts with Categorical column
+        s = ak.DataFrame({"a": ak.Categorical(ak.array(["a", "a", "a", "b"]))}).groupby("a").count()
+        pds = pd.Series(data=np.array([3, 1]), index=pd.Index(data=np.array(["a", "b"], dtype="<U7")))
+        assert_series_equal(pds, s.to_pandas())
+
+    def test_gb_series(self):
+        username = ak.array(["Alice", "Bob", "Alice", "Carol", "Bob", "Alice"])
+        userid = ak.array([111, 222, 111, 333, 222, 111])
+        item = ak.array([0, 0, 1, 1, 2, 0])
+        day = ak.array([5, 5, 6, 5, 6, 6])
+        amount = ak.array([0.5, 0.6, 1.1, 1.2, 4.3, 0.6])
+        bi = ak.arange(2**200, 2**200 + 6)
+        df = ak.DataFrame(
+            {
+                "userName": username,
+                "userID": userid,
+                "item": item,
+                "day": day,
+                "amount": amount,
+                "bi": bi,
+            }
+        )
+
+        gb = df.GroupBy("userName", use_series=True)
+
+        c = gb.count()
+        assert isinstance(c, ak.Series)
+        assert c.index.to_list() == ["Bob", "Alice", "Carol"]
+        assert c.values.to_list() == [2, 3, 1]
+
```

Comment:
We dropped `test_to_pandas`, I just wanna make sure that was intentional

Reply:
Yeah, I don't think its necessary as most of the test cases are using `to_pandas` for verification.

Reply:
I agree, I don't really see the need for that test anymore